vendor/views/vendor.py

```diff
@@ -347,12 +339,12 @@ class OrderHistoryDetailView(LoginRequiredMixin, DetailView):
     slug_url_kwarg = 'uuid'
 
 
-class ProductsListView(LoginRequiredMixin, ListView):
+class ProductsListView(LoginRequiredMixin, SetSiteToRequestMixin, ListView):
     model = Receipt
     template_name = 'vendor/purchase_list.html'
 
     def get_queryset(self):
-        return self.request.user.customer_profile.get(site=get_current_site(self.request)).receipts.filter(status__gte=PurchaseStatus.COMPLETE)
+        return self.request.user.customer_profile.get(site=self.request.site).receipts.filter(status__gte=PurchaseStatus.COMPLETE)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
changed to: `site=get_site_from_request(self.request)`